### DISCUSSION TITLE
Cannot find device "tun_srsue"

### DISCUSSION THREAD

**Discussion by wacky4alanis** [srsran/srsRAN_Project#880] [General Help] [answered]
I am trying to connect srsUE to srsRAN gNB using 2 USRPs.  The srsUE seems to connect OK:

Setting manual TX/RX offset to 100 samples
Waiting PHY to initialize ... done!
Attaching UE...
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=4974
Random Access Complete.     c-rnti=0x4601, ta=0
RRC Connected
PDU Session Establishment successful. IP: 10.45.1.2
RRC NR reconfiguration successful.

But then when I try to follow the instructions to test the connection with a ping, I get this error:

sudo ip route add default via 10.45.1.1 dev tun_srsue
Cannot find device "tun_srsue"

I thought that the srsue program was supposed to create the tun_srsue device when it gets an IP address?  Any idea what I am doing wrong?

**Comment by pgawlowicz** [comment]
yes, the srsUE should create the `tun_srsue`, did you maybe rename it in the ue config?

```
[gw]
#netns =
#ip_devname = tun_srsue
#ip_netmask = 255.255.255.0
```

**Reply by wacky4alanis** [reply]
No - I didn't change that setting.  Here is our entire config file:

[rf]
freq_offset = 0
tx_gain = 0
rx_gain = 0
srate = 30.72e6
nof_antennas = 1

device_name = uhd
device_args = type=x300,addr=192.168.40.63,master_clock_rate=184.32e6,send_frame_size=8000,recv_frame_size=8000,clock=external,sync=external
time_adv_nsamples = 100

[rat.eutra]
dl_earfcn = 2850
nof_carriers = 0

[rat.nr]
bands = 3
nof_carriers = 1
max_nof_prb = 106
nof_prb = 106

[pcap]
enable = none
mac_filename = /tmp/ue_mac.pcap
mac_nr_filename = /tmp/ue_mac_nr.pcap
nas_filename = /tmp/ue_nas.pcap

[log]
all_level = info
phy_lib_level = none
all_hex_limit = 32
filename = /tmp/ue.log
file_max_size = -1

[usim]
mode = soft
algo = milenage
opc  = 63BFA50EE6523365FF14C1F45F88737D
k    = 00112233445566778899aabbccddeeff
imsi = 001010123456780
imei = 353490069873319

[rrc]
release = 15
ue_category = 4

[nas]
apn = srsapn
apn_protocol = ipv4

[gw]
netns = ue1
ip_devname = tun_srsue
ip_netmask = 255.255.255.0

[gui]
enable = false

[expert]
lte_sample_rates = true

**Reply by pgawlowicz** [reply]
ok, could you share logs?

**Comment by wacky4alanis** [comment]
Here is my /tmp/ue.log file - let me know if you'd like to see anything else.
[ue.log](https://github.com/user-attachments/files/17462111/ue.log)

**Reply by pgawlowicz** [reply]
it looks like the interface is setup correctly. but here are some issues:

````
2024-10-18T14:39:46.155044 [STCK   ] [W] Detected slow task processing (sync_queue_len=19).
2024-10-18T14:39:46.155052 [STCK   ] [W] Detected slow task processing (sync_queue_len=18).
2024-10-18T14:39:46.155059 [STCK   ] [W] Detected slow task processing (sync_queue_len=17).
2024-10-18T14:39:46.155066 [STCK   ] [W] Detected slow task processing (sync_queue_len=16).
```

did you run this script on both gnb and srsue PCs?
 https://github.com/srsran/srsRAN_Project/blob/main/scripts/srsran_performance

**Reply by wacky4alanis** [reply]
I did, but unfortunately it failed on the server that is running srsue...  So that is probably my issue?

**Reply by wacky4alanis** [reply]
sudo ./scripts/srsran_performance 
Would you like to set the CPU governor to performance: [Y/n]: Y
tee: '/sys/devices/system/cpu/cpu*/cpufreq/scaling_governor': No such file or directory

**Reply by pgawlowicz** [reply]
which OS do you use? do you run in VM?

**Reply by wacky4alanis** [reply]
Ubuntu 22.04.4 LTS
Not running a VM
Both servers are identical Dell models, running the exact same OS.  The script worked on one, and not the other.  Maybe a BIOS setting?

**Reply by pgawlowicz** [reply]
hmm, strange. could you then run the srsUE on the PC that supports scaling-governonr? just to test whether it works

**Reply by wacky4alanis** [reply, marked as answer]
I was able to change the BIOS settings to match the other machine, and then the srsran_performance script ran fine.   Unfortunately, after doing that, I still had the same error with tun_srsue device not found.  But after some more experimentation, I was able to fix that error by commenting out this line from my ue_rf.conf file:

# netns = ue1

Now the ping is working fine (at least for a while until srsue craps out and gets kicked off - which only takes a minute or 2).  I just need to figure out how to make it work longer without dying...
